src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -515,110 +452,169 @@ public void updateSubscribers()
-
-                        subscribers.stream()
-                                   .filter(player -> citizen.isDirty() || !oldSubscribers.contains(player))
-                                   .forEach(player -> MineColonies.getNetwork().sendTo(msg, player));
-                    }
+                    MineColonies.getNetwork().sendTo(new ColonyViewMessage(this, isNewSubscriber), player);
                 }
             }
+        }
+    }
 
-            //  Buildings
-            if (isBuildingsDirty || hasNewSubscribers)
+    private static boolean hasNewSubscribers(Set<EntityPlayerMP> oldSubscribers, Set<EntityPlayerMP> subscribers)
+    {
+        for (EntityPlayerMP player : subscribers)
+        {
+            if (!oldSubscribers.contains(player))
             {
-                for (Building building : buildings.values())
-                {
-                    if (building.isDirty() || hasNewSubscribers)
-                    {
-                        ColonyViewBuildingViewMessage msg = new ColonyViewBuildingViewMessage(building);
+                return true;
+            }
+        }
+        return false;
+    }
 
-                        subscribers.stream()
-                                   .filter(player -> building.isDirty() || !oldSubscribers.contains(player))
-                                   .forEach(player -> MineColonies.getNetwork().sendTo(msg, player));
-                    }
+    private static boolean isCitizenMissingFromWorld(CitizenData citizen)
+    {
+        EntityCitizen entity = citizen.getCitizenEntity();
+
+        return entity != null && entity.worldObj.getEntityByID(entity.getEntityId()) != entity;
+    }
+
+    private boolean areAllColonyChunksLoaded(TickEvent.WorldTickEvent event)
+    {
+        int distanceFromCenter = Configurations.workingRangeTownHall + 48 /* 3 chunks */ + 15 /* round up a chunk */;
+        for (int x = -distanceFromCenter; x <= distanceFromCenter; x += 16)
+        {
+            for (int z = -distanceFromCenter; z <= distanceFromCenter; z += 16)
+            {
+                if (!event.world.isBlockLoaded(new BlockPos(getCenter().getX() + x, 128, getCenter().getZ() + z)))
+                {
+                    return false;
                 }
             }
         }
+        return true;
+    }
 
-//        for (EntityPlayerMP oldPlayers : oldSubscribers)
-//        {
-//            if (!subscribers.contains(oldPlayers))
-//            {
-//                //  This player should no longer subscribe
-//            }
-//        }
+    private void cleanUpBuildings(TickEvent.WorldTickEvent event)
+    {
+        List<AbstractBuilding> removedBuildings = null;
 
-        isDirty = false;
-        isCitizensDirty = false;
-        isBuildingsDirty = false;
-        permissions.clearDirty();
+        for (AbstractBuilding building : buildings.values())
+        {
+            BlockPos loc = building.getLocation();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'loc' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)